plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/procedure/FlushHiveMetastoreCacheProcedure.java

```diff
@@ -48,15 +65,49 @@ public Procedure get()
     {
         return new Procedure(
                 "system",
-                "flush_metadata_cache",
-                ImmutableList.of(),
+                PROCEDURE_NAME,
+                ImmutableList.of(
```

Comment:
what about `$fake_first_parameter`, per @findepi 's https://github.com/trinodb/trino/pull/10385#issuecomment-1004752358 ?

cc @losipiuk

Reply:
I would rather suggest an explicit boolean flag for procedure definition "allow_only_keword_args".

Reply:
boolean has the diadvantage that you can pass `true` or `false` on the call site and it doesn't scream you're doing something wrong. 
varchar allows you to request a length _and meaningful_ text to be passed in, so of the procedure call is cheated, the call site makes the cheat clear

Reply:
While implementing I was having mixed feeling's about it. 
That's why i left only explicit checks and a meaningful exception.
Resons why:
- it's required IMO to  check if someone actually is not passing this `$fake_first_parameter` (extra check). If it's passed we should throw exception. 
- Also we still need to perform all checks as there is nothing preventing users from passing some subset of given parameters. 
- we allow calling procedure only with named parameters which might be confusing (we will need to leave meaningful description for other developers to understand why it's done that way)

Reply:
> it's required IMO to check if someone actually is not passing this $fake_first_parameter (extra check). If it's passed we should throw exception.

yes, that's the intended use of the parameter.

> we allow calling procedure only with named parameters which might be confusing 

this is the goal, and the gain

this allows us to modify procedure signature, with less backwards compatibility worries

Reply:
> boolean has the diadvantage that you can pass true or false on the call site and it doesn't scream you're doing something wrong.

@findepi  I meant extending `Procedure` class with field:
```
private final boolean requireKeywordArguments;
```

And adding support for that in `CallTask`

Reply:
Ok added.

Reply:
@findepi @losipiuk I've added solution proposed by @losipiuk in separate commits for now.
If it will be ok I will reorganise and squash them

Reply:
Ok will move it to separate PR